app/controllers/portal/locker/locker_ballots_controller.rb

```diff
@@ -1,75 +1,76 @@
+        @round = LockerRound.active_at(current_time).first
+        if @round && current_user
+            @ballot = LockerBallot.where('user_id = ? and locker_round_id = ?', current_user.id, @round.id).first_or_initialize
+        end
     end
-  end
 
-	def create
-		@ballot = LockerBallot.new ballot_params	
-		@ballot.user_id = current_user.id
+    def create
+        @ballot = LockerBallot.new ballot_params	
+        @ballot.user_id = current_user.id
 
-		respond_to do |format|
-			if DateTime.now <= @ballot.round.end && @ballot.save
-        mail_notify(current_user, @ballot)
-				format.html {
-          redirect_to portal_locker_locker_ballots_path,
-                      notice: 'Ballot application submitted! Please check your NUS email for confirmation. Do contact us at connect@nuscomputing.com if there are any issues.'
-				}
-				format.json { render json: @ballot, status: :created, location: @ballot }
-			else
-				format.html { redirect_to portal_locker_locker_ballots_path }
-				format.json { render json: @ballot.errors, status: :unprocessable_entity }
-			end
-		end
-	end
+        respond_to do |format|
+            if DateTime.now <= @ballot.round.end && @ballot.save
+                mail_notify(current_user, @ballot)
+                format.html {
+                    redirect_to portal_locker_locker_ballots_path,
+                    notice: 'Ballot application submitted! Please check your NUS email for confirmation. Do contact us at connect@nuscomputing.com if there are any issues.'
+                }
+                format.json { render json: @ballot, status: :created, location: @ballot }
+            else
+                format.html { redirect_to portal_locker_locker_ballots_path }
+                format.json { render json: @ballot.errors, status: :unprocessable_entity }
+            end
+        end
+    end
 
-	def update
-    if is_authenticated_user
-      if @ballot.update(ballot_params)
-        mail_notify(current_user, @ballot)
-        redirect_to portal_locker_locker_ballots_path,
+    def update
+        if is_authenticated_user
+            if @ballot.update(ballot_params)
+                mail_notify(@ballot.user, @ballot)
+                redirect_to portal_locker_locker_ballots_path,
                     notice: 'Ballot updated! Please check your NUS email for confirmation. Do contact us at connect@nuscomputing.com if there are any issues.'
-      end
-    else
-      show_error_message_at portal_locker_locker_ballots_path
+            end
+        else
+            show_error_message_at portal_locker_locker_ballots_path
+        end
     end
-	end
 
-	def destroy
-		if is_authenticated_user
-      @ballot.destroy
-      redirect_to portal_locker_locker_ballots_path, notice: 'Your Ballot has been cancelled'
-    else
-      show_error_message_at portal_locker_locker_ballots_path
+    def destroy
+        if is_authenticated_user
+            mail_notify(@ballot.user, @ballot)
```

Comment:
More changes to mailers/ballot_notifier.rb and its views are probably required to fix #169. Current changes will send an email telling users a ballot was submitted when it has been deleted.